edgedb/scram/__init__.py

```diff
@@ -187,16 +187,18 @@ def parse_client_first_message(resp: bytes):
     if nonce_attr[0:1] != b'r':
         raise ValueError('malformed SCRAM message')
 
-    _, _, nonce_b64 = nonce_attr.partition(b'=')
-    nonce = base64.b64decode(nonce_b64)
+    _, _, nonce_bin = nonce_attr.partition(b'=')
+    nonce = nonce_bin.decode('ascii')
+    if not nonce.isprintable():
```

Comment:
If you'd like to be thorough, reject commas here, too.

Reply:
The original string is split by commas, so commas are not possible here.